pkgs/development/python-modules/einops/default.nix

```diff
@@ -26,38 +31,47 @@ buildPythonPackage rec {
   };
 
   checkInputs = [
+    chainer
+    jupyter
+    keras
+    # mxnet (has issues with some CPUs, segfault)
+    nbconvert
+    nbformat
     nose
     numpy
-    # For notebook tests
-    nbformat
-    nbconvert
-    jupyter
-    # For backend tests
-    chainer
     parameterized
+    pytestCheckHook
     pytorch
-    mxnet
     tensorflow
-    keras
   ];
 
   # No CUDA in sandbox
   EINOPS_SKIP_CUPY = 1;
 
-  checkPhase = ''
-    export HOME=$TMPDIR
-
+  preCheck = ''
+     export HOME=$(mktemp -d);
+    
     # Prevent hangs on PyTorch-related tests, see
     # https://discuss.pytorch.org/t/pytorch-cpu-hangs-on-nn-linear/17748/4
-    export OMP_NUM_THREADS=1
-
-    nosetests -v -w tests
```

Comment:
Why removing `nosetests` in favor of `pytest`? The upstream recommends `nose` (also it is still in the argument list).

Reply:
`nose` is unmaintained for 7 years now and the "pytest eco-system" in nixpkgs has features which makes the handling of tests simpler.

Reply:
Seems fair then.